### PR TITLE
Add missing examples in list

### PR DESCRIPTION
**Pull request by gfcas**
Related to #815

This patch adds the three missing examples into the list. Any suggestions are welcome. For the brief description I took the headlines of the corresponding examples. Is the ordering in the list OK?

### CODE REVIEW

**Comment by gfcas** [comment]
What do you think of an extra topic 'mixed precision' in the Advanced Techniques section at the bottom of the page?

**Comment by yhmtsai** [comment]
Thank you! @gfcas 
Looks like we miss many examples in the documentation.
Could you also add these into Examples?
```
cb-gmres
heat-equation
ir-ilu-preconditioned-solver
iterative-refinement
par-ilu-convergence
preconditioner-export
```
I also compiled this document and gives the following picture
image

**Comment by gfcas** [comment]
@yhmtsai Of course, no problem. The missing ones are now added. In addition I opened a topic 'Iterative refinement' under basic techniques, as the iterative_refinement example is marked as such a one. If you prefer further topics, let me know. Maybe with a small instruction (topic name and correspondig examples).

**Review by yhmtsai — APPROVED** [review]
LGTM. maybe the group will be changed after some discussion but I think it is good now

**Comment by yhmtsai** [comment]
close #819

File: doc/examples/examples.hpp.in [hunk 1]
```diff
@@ -187,6 +187,59 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+ *
+ *   <tr valign="top">
+ *       <td>@ref cb_gmres</td>
+ *       <td> Using the Ginkgo CB-GMRES solver (Compressed Basis GMRES).
+ *       </td></tr>
+ *
+ *   <tr valign="top">
+ *       <td>@ref heat_equation</td>
+ *       <td> Solving a 2D heat equation and showing matrix assembly, vector
+ *            initalization and solver setup in a more complex setting.
```
Comment (pratikvn) [inline doc/examples/examples.hpp.in:215]: Maybe you can also add that there is some visualization here.

File: doc/examples/examples.hpp.in [hunk 2]
```diff
@@ -187,6 +187,59 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+ *       </td></tr>
+ *
+ *   <tr valign="top">
+ *       <td>@ref iterative_refinement</td>
+ *       <td> Using an inaccurate CG solver as inner solver to an iterative
```
Comment (pratikvn) [inline doc/examples/examples.hpp.in:220]: nit:
```suggestion
 *       <td> Using an low accuracy CG solver as inner solver to an iterative
```

File: doc/examples/examples.hpp.in [hunk 3]
```diff
@@ -225,6 +278,14 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  *   </tr>
  *
  *   <tr valign="top">
+ *     <td> Iterative refinement
```
Comment (pratikvn) [inline doc/examples/examples.hpp.in:281]: I would also add `mixed_precision_ir` here.

**Review by pratikvn — APPROVED** [review]
Thanks for adding these! LGTM!

File: doc/examples/examples.hpp.in [hunk 4]
```diff
@@ -187,6 +187,59 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+ *   <tr valign="top">
+ *       <td>@ref mixed_precision_ir</td>
+ *       <td> Manual implementation of a Mixed Precision Iterative Refinement
+ *            (MPIR) solver.
+ *       </td></tr>
+ *
+ *   <tr valign="top">
+ *       <td>@ref adaptiveprecision_blockjacobi</td>
+ *       <td> Shows how to use the adaptive precision block-Jacobi
+ *            reconditioner.
```
Comment (tcojean) [inline doc/examples/examples.hpp.in:204]:
```suggestion
 *            preconditioner.
```

File: doc/examples/examples.hpp.in [hunk 5]
```diff
@@ -225,6 +278,14 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  *   </tr>
  *
  *   <tr valign="top">
+ *     <td> Iterative refinement
+ *     </td>
+ *     <td>@ref iterative_refinement,
+ *         @ref ir_ilu_preconditioned_solver
```
Comment (tcojean) [inline doc/examples/examples.hpp.in:284]: This example can also be added to `Using preconditioners` a few lines above.
For the same category, we can add:
`adaptiveprecision_blockjacobi`
`par_ilu_convergence`
`preconditioner_export`.

File: doc/examples/examples.hpp.in [hunk 6]
```diff
@@ -196,14 +249,14 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
  *     <td>@ref simple_solver
  *     </td>
  *   </tr>
  *
  *   <tr valign="top">
- *     <td> Debug the performance of a solver using loggers.
+ *     <td> Debug the performance of a solver using loggers
  *     </td>
  *     <td>@ref performance_debugging
  *     </td>
```
Comment (tcojean) [inline doc/examples/examples.hpp.in:262]: Maybe we can add `preconditioner_export` here? It's not strictly performance debugging, but it's debugging for sure.
```suggestion
 *     <td> Debug the performance of a solver or preconditioner.
 *     </td>
 *     <td>@ref performance_debugging
 *         @ref preconditioner_export
 *     </td>
```

File: doc/examples/examples.hpp.in [hunk 7]
```diff
@@ -278,14 +339,14 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
  *     <td> @ref inverse_iteration
  *     </td>
  *   </tr>
  *
  *   <tr valign="top">
- *     <td> Logging within Ginkgo.
+ *     <td> Logging within Ginkgo
  *     </td>
  *     <td> @ref simple_solver_logging,
  *          @ref papi_logging,
```
Comment (tcojean) [inline doc/examples/examples.hpp.in:352]:
```suggestion
 *          @ref papi_logging,
 *          @performance_debugging
```

**Review by tcojean — APPROVED** [review]
LGTM! Thanks a lot for this useful work. Here are a few suggestions.

File: doc/examples/examples.hpp.in (hunk 5 above)
Comment (gfcas) [reply doc/examples/examples.hpp.in:284]: Both, ie ``iterative_refinemen`` and ``ir_ilu_preconditioned_solver`` to ``Using preconditioners``?

File: doc/examples/examples.hpp.in [hunk 8]
```diff
@@ -187,6 +187,60 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+ *       </td></tr>
+ *
+ *   <tr valign="top">
+ *       <td>@ref iterative_refinement</td>
+ *       <td> Using a low accuracy CG solver as an inner solver to an iterative
+ *            refinement (IR) method which solves a linear system.
+ *       </td></tr>
+ *
+ *   <tr valign="top">
+ *       <td>@ref  ir_ilu_preconditioned_solver</td>
```
Comment (yhmtsai) [inline doc/examples/examples.hpp.in:226]: nit
```suggestion
 *       <td>@ref ir_ilu_preconditioned_solver</td>
```

Comment (gfcas) [reply doc/examples/examples.hpp.in]: Thanks!

File: doc/examples/examples.hpp.in (hunk 5 above)
Comment (yhmtsai) [reply doc/examples/examples.hpp.in:291]: no for iterative_refinement but yes for ir_ilu_preconditioned_solver, I think

Comment (gfcas) [reply doc/examples/examples.hpp.in:291]: ``ir_ilu_preconditioned_system`` is already added. I was not sure about ``iterative_refinement``. Fine. Thanks!

**Pull request merged**